features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/processor/RecordedDataProcessor.kt

```diff
@@ -24,22 +24,22 @@ internal class RecordedDataProcessor(
     private val mutationResolver: MutationResolver,
     private val nodeFlattener: NodeFlattener = NodeFlattener()
 ) : Processor {
-
     private var prevSnapshot: List<MobileSegment.Wireframe> = emptyList()
     private var lastSnapshotTimestamp = 0L
     private var previousOrientation = Configuration.ORIENTATION_UNDEFINED
+    private var prevRumContext: SessionReplayRumContext = SessionReplayRumContext()
```

Comment:
I'm not quite clear what this change does: if `RecordedDataProcessor#processScreenSnapshots` is called always from the same thread, then it shouldn't be the difference before and after this change. If it is called from multiple threads, then how do we avoid concurrency issues here?

 Also is there any impact on storing this as a property from the fact that we have early return in `RecordedDataProcessor#handleSnapshots` (when wireframes are empty).

Reply:
Thanks for the explanation!

> It's called on a synchronized method that launches a new background thread each time (although the threadpool size is 1, so at most one thread alive at any time)

`RecordedDataQueueHandler` has a `ThreadPoolExecutor` as executor service, this implementation won't create a **new** thread each time, it will have one and the same thread running always unless it dies - in this case it will create a new one.

Anyway, it won't change anything, because this method is called from the synchronised one.